### DISCUSSION TITLE
Can I resuse common repostitories across different project?

### DISCUSSION THREAD

**Discussion by demonguy** [oracle/opengrok#4484] [General]
Suppose I have to depoly 5 projects on opengrok.
Each project cotain a common repository A, and a project specific repository.

In order to deploy 5 projects. I have to download common repository 5 times, and index it 5 times.
Is there anyway to download common repository 1 times and makes it usable for all 5 projects?

**Comment by vladak** [comment]
Would symlinking the common repositories work ? If not then you can try cloning the shared repositories locally from one that is mirrored remotely you may save some mirroring time (assuming this is Git or such that has fast local clones). Storage savings wise, a file-system with built-in compression might help.

**Reply by demonguy** [reply]
Does it mean at least we still need to index 5 times for the common repository?
Is it possible to index  the common repository and make it usable for all 5 projects?

**Reply by vladak** [reply]
> Does it mean at least we still need to index 5 times for the common repository?

yes.

> Is it possible to index  the common repository and make it usable for all 5 projects?

That's not possible.
